### DISCUSSION TITLE
Want to use AWS s3 for working dir

### DISCUSSION THREAD

**Discussion by snowhork** [open-mmlab/mmclassification#1110] [Q&A] [answered]
### Prerequisite

- [X] I have searched related issues but cannot get the expected help.
- [X] I have read [the documentation](https://mmclassification.readthedocs.io/) and don't know what to do.

### Describe the question

Hi, Dear.

I want to use AWS s3 bucket for working directory. 

* Is it already implemented? (maybe not I guess)
* Any best practice?
* Can I override working directory behaviour?

Thanks!

**Comment by mzr1996** [comment, marked as answer]
Not yet.
Usually, MMClassification doesn't handle file io related functions, and we use the `FileClient` in mmcv to deal with different file backends. And the AWS backend support is still in progress, see https://github.com/open-mmlab/mmcv/pull/1545.
Of course, you can also implement a data transform to load images from AWS, refers to [LoadImageFromFile](https://mmclassification.readthedocs.io/en/master/_modules/mmcls/datasets/pipelines/loading.html#LoadImageFromFile)

**Comment by snowhork** [comment]
@mzr1996 

Thanks for quick reply! Your information is very convenient for me.

**Comment by zhouzaida** [comment]
Hi @snowhork , the AWS backend will be provided in [MMEngine](https://github.com/open-mmlab/mmengine) in the next month.

**Reply by snowhork** [reply]
@zhouzaida Good news! Any related issue or PR?

**Reply by zhouzaida** [reply]
Hi, we will migrate the implementation in https://github.com/open-mmlab/mmcv/pull/1545 to MMEngine.
